File: backend/python/src/routing/router.py
#!/usr/bin/env python3
"""Request Routing"""

from typing import Dict, Callable

class Router:
    def __init__(self):
        self.routes = {}
    
    def add_route(self, path: str, handler: Callable):
        self.routes[path] = handler
    
    def route(self, request: Dict):
        path = request.get("path", "")
        handler = self.routes.get(path)
        
        if not handler:
            return {"status": 404, "body": "Not Found"}
        
        return handler(request)
    
    def list_routes(self):
        return list(self.routes.keys())

def health_handler(req):
    return {"status": 200, "body": "OK"}

router = Router()
router.add_route("/health", health_handler)
print(router.route({"path": "/health"}))

File: backend/ruby/lib/matching/router.rb
module TigerEx
  module Matching
    class Router
      def initialize
        @routes = {}
      end
      
      def add_route(path, handler)
        @routes[path] = handler
      end
      
      def route(request)
        path = request[:path]
        handler = @routes[path]
        
        if handler.nil?
          return { status: 404, body: "Not Found" }
        end
        
        handler.call(request)
      end
      
      def list_routes
        @routes.keys
      end
    end
  end
end